test/utils/base/cluster_context.go

```diff
@@ -47,6 +48,18 @@ func (cc *ClusterContext) KubectlExec(command string) ([]byte, error) {
 }
 
 func (cc *ClusterContext) CreateNamespace() error {
+	if ShouldSkipNamespaceSetup() {
+		log.Printf("Skipping namespace creation for %v", cc.Namespace)
+		ns, err := cc.VanClient.KubeClient.CoreV1().Namespaces().Get(cc.Namespace, metav1.GetOptions{})
+		if err == nil && ns != nil {
+			// As we're skipping the creation of namespaces, we're adopting whatever
+			// we find; we will destroy these when DeleteNamespace is called, unless
+			// ShouldSkipNamespaceTeardown returns true.
+			log.Printf("Reusing existing namespace %v", cc.Namespace)
+			cc.nsCreated = true
+		}
+		return nil
```

Comment:
I would go with: "Return an error" (fail fast). Although I don't know what is the benefit of using it,
since if a namespace already exists, it probably has stuff running in it and that stuff may interfere
in the test execution.